transcom/mymove
# Int b 21373 validation submit move details task order with files

## [B-21373](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1014638&RoomContext=TeamRoom%3A1007470&concept=TeamRoom)

## Summary
This pr is to add disabling of the "Submit move details" and "Approve selected" buttons on the Move details page. It also adds and alert message /disabling for "Manage orders" /"Done buttons if there's no uploaded documents. A scroll to view is also added to focus on required fields when data is not entered into a required field. 
Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the application
2. Login as a SC 
3. Search for a Move (could be new or already created), navigate to the Move details page.
4. Scroll down to "Orders" section and click the "View and edit orders" button.
5. "Manage orders" and "Done" buttons should be disabled with a message "File upload is required" if there's no document uploaded. 
6. Click on the "Save" button, should stay on the same page and receive "Please select one or more files"
7. Upload a document and click the Save button.
8. Should be navigated to the "Move details" page with "Submit move details" button enabled (if all required orders fields are entered).
9. Verify disabling of the Submit move details button by repeating step 4 and then deleting the uploaded document and click the "Cancel" button. When navigated back to the Move details page the Submit move details button should be disabled.
10. Login as a TOO
11. Repeat step 3 - 7
12. Should be navigated to the "Move details" page, select a shipment(Move management seems to be selected by default) and the "Approve selected" button should be enabled. 
13. Verify disabling of the Approve selected button by repeating step 4 and then deleting the uploaded document and click the "Cancel" button.
14. Repeat step 12 and the "Approve selected button should be disabled.
15. Login as SC/TOO
16. Repeat step 3 and 4
17. Make sure some of the required fields are not filled out(e.g Date issued, Report by date, Department indicator, Orders number) and scroll out of focus any of the fields not entered.
18. Click on the "Save" button, should scroll to focus the first required field not entered. 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots

https://github.com/user-attachments/assets/7e6a70d3-fc1c-4480-9b05-40fb131d1f06


https://github.com/user-attachments/assets/11e8831b-d502-4304-a2bc-0bbb7bbfd779


https://github.com/user-attachments/assets/b0d47815-b62e-4145-9ff9-8c52873543a0
